docs/api.rst

```diff
@@ -13866,3 +13866,98 @@ Managing calendar reminders
   :statuscode 200: Data queried correctly.
   :statuscode 401: No user is currently logged in.
   :statuscode 500: Internal rotki error.
+
+Historical Balance Queries
+==============================
+
+.. http:put:: /api/(version)/balances/historical
+
+  Gets a snapshot of asset balances at a specific timestamp, calculated from processing of historical events.
+
+  Example Request:
+
+  .. http:example:: curl wget httpie python-requests
+
+    PUT /api/(version)/balances/historical HTTP/1.1
+    Host: localhost:5042
+    Content-Type: application/json;charset=UTF-8
+
+    { "timestamp": 1672531200 }
+
+  Example Response:
+
+  .. sourcecode:: http
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json
+
+    {
+      "message": "",
+      "result": {
+        "BTC": {
+          "amount": "2.0",
+          "price": "20000"
+        },
+        "ETH": {
+          "amount": "10.0",
+          "price": "1500"
+        }
+      },
+      "status_code": 200
+    }
+
+  :resjson object result: Object mapping asset identifiers to their balance info
+  :resjson string amount: The amount of the asset held at the timestamp
+  :resjson string price: The price of the asset at the timestamp in the user's profit currency
+  :statuscode 200: Historical balances returned
+  :statuscode 400: Malformed query
+  :statuscode 404: No historical data found for the timestamp
+  :statuscode 403: User does not have premium access
+  :statuscode 409: User is not logged in
+  :statuscode 500: Internal Rotki error
+
+
+.. http:post:: /api/(version)/balances/historical
```

Comment:
Looking at this ... would it not make sense to combine those too then? If asset is missing give snapshot of all, if asset is given only of that asset? And keep it in the `POST` verb?